pkg/insight/insightstore/deployment_store.go

```diff
@@ -0,0 +1,338 @@
+// Copyright 2022 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightstore
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipecd/pkg/filestore"
+	"github.com/pipe-cd/pipecd/pkg/insight"
+)
+
+type DeploymentBlockMetadata struct {
+	BlockID       string                     `json:"block_id"`
+	ChunkMetadata []*DeploymentChunkMetadata `json:"chunk_metadata"`
+}
+
+type DeploymentChunkMetadata struct {
+	ChunkID      string `json:"chunk_id"`
+	ChunkIndex   int    `json:"chunk_index"`
+	MinTimestamp int64  `json:"min_timestamp"`
+	MaxTimestamp int64  `json:"max_timestamp"`
+	Count        int    `json:"count"`
+	Completed    bool   `json:"completed"`
+}
+
+type DeploymentChunk struct {
+	ChunkID     string                    `json:"chunk_id"`
+	Deployments []*insight.DeploymentData `json:"deployments"`
+}
+
+func (c *DeploymentChunk) Size() (int64, error) {
+	raw, err := json.Marshal(c)
+	if err != nil {
+		return 0, err
+	}
+
+	return int64(len(raw)), nil
+}
+
+func (c *DeploymentChunk) FindDeployments(from, to int64) []*insight.DeploymentData {
+	out := make([]*insight.DeploymentData, 0, len(c.Deployments))
+	for _, d := range c.Deployments {
+		if from <= d.CompletedAt && d.CompletedAt <= to {
+			out = append(out, d)
+		}
+	}
+	return out
+}
+
+func (m *DeploymentBlockMetadata) FindChunks(from, to int64) []string {
+	var ids []string
+	for _, m := range m.ChunkMetadata {
+		if overlap(from, to, m.MinTimestamp, m.MaxTimestamp) {
+			ids = append(ids, m.ChunkID)
+		}
+	}
+	return ids
+}
+
+func overlap(lhsFrom, lhsTo, rhsFrom, rhsTo int64) bool {
+	return (rhsFrom < lhsTo && lhsFrom < rhsTo)
```

Comment:
```suggestion
	return (rhsFrom < lhsTo || lhsFrom < rhsTo)
```
should this be like this? or
```suggestion
	return (lhsFrom <= rhsFrom && lhsTo <= rhsTo)
```

Reply:
@khanhtc1202 Nice catch! That was an old function. I have just addressed it by https://github.com/pipe-cd/pipecd/pull/3993/commits/acd28597d17cc7d9b43a5c4461aeb82896a332a2.